File: src/main/org/nlogo/window/SliderPainter.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window

import java.awt.{ Dimension, Font, Graphics }

abstract class SliderPainter {
  def getMinimumSize: Dimension
  def getPreferredSize(font: Font): Dimension
  def getMaximumSize: Dimension
  def setToolTipText(text: String): Unit
  def doLayout(): Unit
  def paintComponent(g: Graphics): Unit
  def dettach(): Unit
}

File: src/main/org/nlogo/window/SliderPainter.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window;

public abstract strictfp class SliderPainter {
  abstract public java.awt.Dimension getMinimumSize();

  abstract public java.awt.Dimension getPreferredSize(java.awt.Font font);

  abstract public java.awt.Dimension getMaximumSize();

  abstract public void doLayout();

  abstract public void paintComponent(java.awt.Graphics g);

  abstract public void dettach();
}
